pkg/commands/app.go

```diff
@@ -1232,6 +1234,62 @@ func NewCleanCommand(globalFlags *flag.GlobalFlagGroup) *cobra.Command {
 	return cmd
 }
 
+func NewAuthCommand(globalFlags *flag.GlobalFlagGroup) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "auth [flags]",
+		GroupID:       groupUtility,
+		Short:         "Authentication",
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	loginFlags := &flag.Flags{
+		GlobalFlagGroup:   globalFlags,
+		RegistryFlagGroup: flag.NewRegistryFlagGroup(),
+	}
+	loginFlags.RegistryFlagGroup.RegistryToken = nil // disable '--registry-token'
+	loginCmd := &cobra.Command{
+		Use:           "login SERVER",
+		Short:         "Log in to a registry",
+		SilenceErrors: true,
+		SilenceUsage:  true,
+		Example: `  # Log in to reg.example.com
+  cat ~/my_password.txt | trivy auth login --username foo --password-stdin reg.example.com`,
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := loginFlags.Bind(cmd); err != nil {
+				return xerrors.Errorf("flag bind error: %w", err)
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			loginOpts, err := loginFlags.ToOptions(args)
+			if err != nil {
+				return xerrors.Errorf("flag error: %w", err)
+			}
+			return auth.Login(cmd.Context(), args[0], loginOpts)
```

Comment:
nit:
`docker logic` can be used without registry (default `docker.index.io` is used in this case).
I guess we are not docker, so we don't need to use the same logic.

But maybe we want to write in docs that default registry for `docker` is `docker.index.io` (because I never thought about it and checked `docker/cli` to find it 😄 ).

Reply:
There are several tools supporting authentication nowadays. Most of them don't have the default registry.

```
$ oras login
Error: "oras login" requires exactly 1 argument but got 0
Usage: oras login [flags] <registry>
Please specify exactly 1 argument as the registry to log in to. Run "oras login -h" for more options and examples
```

```
$ crane auth login
Error: accepts 1 arg(s), received 0
```

```
$ podman login
Error: no registries found in registries.conf, a registry must be provided
```

```
$ trivy auth login
2024-10-08T11:38:31+04:00       FATAL   Fatal error     accepts 1 arg(s), received 0
```

I don't think we need to mention `docker login` here. It's well described.

```
$ trivy auth login -h
Log in to a registry

Usage:
  trivy auth login SERVER [flags]

Examples:
  # Log in to reg.example.com
  cat ~/my_password.txt | trivy auth login --username foo --password-stdin reg.example.com
```

If we see users who get confused, we can update the doc.